project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/GitWrapperService.kt

```diff
@@ -0,0 +1,127 @@
+package org.plan.research.minimization.plugin.services
+
+import  org.plan.research.minimization.plugin.model.IJDDContext
+
+import com.intellij.openapi.application.readAction
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.isProjectOrWorkspaceFile
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.toNioPathOrNull
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.ResetCommand
+import org.eclipse.jgit.api.errors.NoHeadException
+
+import java.io.File
+import java.util.*
+
+import kotlin.io.path.pathString
+import kotlin.io.path.relativeTo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * Service responsible for git operations within ProjectGitSnapshotManager.
+ */
+@Service(Service.Level.APP)
+class GitWrapperService {
+    private val importantFiles = setOf("modules.xml", "misc.xml", "libraries")
```

Comment:
Let's extract it to the separate object for the sake of compatibility and extensibility with cloning